src/mod_stream_management.erl

```diff
@@ -59,19 +73,43 @@ sm() ->
     #xmlel{name = <<"sm">>,
            attrs = [{<<"xmlns">>, ?NS_STREAM_MGNT_3}]}.
 
+-spec remove_smid(Acc, SID, JID, Info, Reason) -> Acc1 when
+      Acc :: mongoose_acc:t(),
+      SID :: ejabberd_sm:sid(),
+      JID :: undefined | jid:jid(),
+      Info :: undefined | [any()],
+      Reason :: undefined | ejabberd_sm:close_reason(),
+      Acc1 :: mongoose_acc:t().
 remove_smid(Acc, SID, _JID, _Info, _Reason) ->
-    case mnesia:dirty_index_read(sm_session, SID, #sm_session.sid) of
+    H = mongoose_acc:get(stream_mgmt, h, undefined, Acc),
+    MaybeSMID = case mnesia:dirty_index_read(sm_session, SID, #sm_session.sid) of
```

Comment:
It doesn't look good to me. If there is no such session we use value `ok` otherwise use the real `SMID`. How about converting this code into a function and returning `{error, not_found}` and `{ok, SMID}` and setting these in the Acc?

Reply:
Great idea 🙂